File: brixen-java/src/main/java/org/brixen/bean/ContentContainerBean.java
package org.brixen.bean;

import org.openqa.selenium.WebElement;

import javax.annotation.ParametersAreNonnullByDefault;

/**
 * Defines the contract for a data transfer object used to construct a <b>Selenium</b> page object that models a
 * component which wraps a {@link WebElement} content container.
 */
@SuppressWarnings("UnusedDeclaration")
@ParametersAreNonnullByDefault
public interface ContentContainerBean extends LoadableBean {
    /**
     * Sets the {@code WebElement} that contains the content wrapped by the page object.
     *
     * @param contentContainer    the {@code WebElement} that contains the content wrapped by the page object
     */
    @SuppressWarnings("NullableProblems")
    void setContentContainer(WebElement contentContainer);

    /**
     * Returns the {@code WebElement} that contains the content wrapped by the page object.
     *
     * @return  the {@code WebElement} that contains the content wrapped by the page object
     */
    WebElement getContentContainer();
}

File: brixen-dotnet/src/bean/ContentContainerBean.cs
using System;
using OpenQA.Selenium;

namespace Org.Brixen.Bean {
	public class ContentContainerBean : LoadableBean, IContentContainerBean
	{
		IWebElement contentContainer;

		public ContentContainerBean() : base() { }

		public IWebElement ContentContainer {
			get {
				return contentContainer;
			}

			set {
				if(value == null) {
					throw new ArgumentNullException ("value", "Cannot invoke setter for ContentContainer property " + 
						"with a null parameter");
				}	

				contentContainer = value;
			}
		}
	}
}

